### PR TITLE
LibPDF: Improve text rendering; PDFViewer: Use ArgsParser

### PR DESCRIPTION
**Pull request by mattco98**
LibPDF:

Lots of text-related changes here, mostly focused around fonts. Enjoy the following screenshots from the PDF 1.7 spec (compared to Firefox) which demonstrate how big of a difference this makes!

<details>
<summary>Page 1: </summary>
<br />

</details>

<details>
<summary>Page 2: </summary>
<br />

</details>

PDFViewer: 

Use ArgsParser to avoid Shell completion shenanigans.

### CODE REVIEW

File: Userland/Applications/PDFViewer/main.cpp [hunk 1]
```diff
@@ -6,25 +6,30 @@
...
-#include <LibGUI/MessageBox.h>
 #include <LibGUI/Window.h>
 #include <LibMain/Main.h>
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    auto app = GUI::Application::construct(arguments);
+    TRY(Core::System::pledge("stdio recvfd sendfd rpath unix"));
+
+    auto app = TRY(GUI::Application::try_create(arguments));
```
Comment (alimpfard) [inline Userland/Applications/PDFViewer/main.cpp:22]: I'm surprised this does anything :thinking:
at this point any pledge call will pretend to succeed, but should only give you `stdio rpath`...unless my impl of `no_error` is flawed

File: Userland/Applications/PDFViewer/main.cpp [hunk 2]
```diff
@@ -6,25 +6,30 @@
...
 
     auto window = GUI::Window::construct();
     window->set_title("PDF Viewer");
     window->resize(640, 400);
 
-    TRY(Core::System::pledge("stdio recvfd sendfd rpath unix"));
+    const char* file_path = nullptr;
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(file_path, "PDF file to open", "path", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
```
Comment (alimpfard) [inline Userland/Applications/PDFViewer/main.cpp:32]: move this to the top and keep the pledge where it was, that should "fix" autocomplete in shell until we figure out a way to either silence it or detect it ahead of time.

File: Userland/Applications/PDFViewer/main.cpp (hunk 1 above)
Comment (alimpfard) [reply Userland/Applications/PDFViewer/main.cpp:22]: yep, that was an impl error: https://github.com/SerenityOS/serenity/pull/13327

**Comment by mattco98** [comment]
Well unfortunately I've had to get rid of the `Fonts/` directory. For some reason, when running `TestPDF`, all of the function in there would refuse to link, even though they are listed in the `CMakeLists.txt` (and it works totally fine when launching Serenity normally). If anyone knows why, please let me know. Otherwise I guess I'll just have everything in one giant folder :/.

**Comment by trflynn89** [comment]
@mattco98 Your linker errors were on Lagom, which does not read `Userland/Libraries/LibPDF/CMakeLists.txt` - that file is used only for Serenity builds.

On Lagom, we currently just basically wildcard source directories. See here:
https://github.com/SerenityOS/serenity/blob/master/Meta/Lagom/CMakeLists.txt#L398=

So to add a subdirectory, you need to add lines to wildcard the subdirectory. See how it's done for LibJS here as an example:
https://github.com/SerenityOS/serenity/blob/master/Meta/Lagom/CMakeLists.txt#L375-L377=

**Comment by mattco98** [comment]
Ahh, that makes sense. I knew Lagom was it's own thing, but I didn't know about that top level CMakeLists. I was only looking at the one in `Tests/LibPDF`. Thanks!

**Pull request merged**